core/dbt/deps/tarball.py

```diff
@@ -42,23 +48,31 @@ def nice_version_name(self):
         return f"tarball (url: {self.tarball})"
 
     def _fetch_metadata(self, project, renderer):
-        """
-        recycle RegistryPackageMetadata so that we can use the install and
-        download_and_untar from RegistryPinnedPackage next.
-        build RegistryPackageMetadata from info passed via packages.yml since no
-        'metadata' service exists in this case.
-        """
-
-        dct = {
-            "name": self.package,
-            "packages": [],  # note: required by RegistryPackageMetadata
```

Comment:
An alternative approach, more in keeping with the previous implementation: Allow users to manually specify the dependent packages that need to be installed.
```yml
# packages.yml
packages:
  - tarball: "https://github.com/calogica/dbt-expectations/archive/refs/tags/0.9.0.tar.gz"
    name: "dbt_expectations"
    # manually specified
    packages:
      - package: calogica/dbt_date
        version: [">=0.9.0", "<1.0.0"]
```
But that doesn't feel any better than just manually specifying both packages:
```yml
# packages.yml
packages:
  - tarball: "https://github.com/calogica/dbt-expectations/archive/refs/tags/0.9.0.tar.gz"
    name: "dbt_expectations"
  - package: calogica/dbt_date
    version: [">=0.9.0", "<1.0.0"]
```